### PR TITLE
Do not crash when a skeleton key is used on an unknown node

### PR DESCRIPTION
**Pull request by Montandalar**

### CODE REVIEW

**Comment by sofar** [comment]
Sure, yeah

File: mods/default/craftitems.lua [hunk 1]
```diff
@@ -160,8 +160,10 @@ minetest.register_craftitem("default:skeleton_key", {
 			return itemstack
 		end
 
-		local on_skeleton_key_use = minetest.registered_nodes[node.name].on_skeleton_key_use
-		if not on_skeleton_key_use then
+		local on_skeleton_key_use
+		if minetest.registered_nodes[node.name] and minetest.registered_nodes[node.name].on_skeleton_key_use then
```
Comment (sfan5) [inline mods/default/craftitems.lua:164]: It would be better to factor out `minetest.registered_nodes[node.name]` to a local variable here

Comment (Montandalar) [reply mods/default/craftitems.lua]: Addressed

Comment (Montandalar) [reply mods/default/craftitems.lua]: woops and once more for good measure (I accidentally used spaces instead of tabs)

**Comment by SmallJoker** [comment]
Shortening it further is possible:

```Lua
		local node_reg = minetest.registered_nodes[node.name]
		local on_skeleton_key_use = node_reg and node_reg.on_skeleton_key_use
		if not on_skeleton_key_use then
			return itemstack
		end
```

**Comment by Montandalar** [comment]
Golfed as per your request

**Pull request merged**